Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2229,7 +2248,9 @@ public extension Tensor where Scalar: Numeric {
         exclusive: Bool = false,
         reverse: Bool = false
     ) -> Tensor {
-        _Raw.cumsum(self, axis: axis, exclusive: exclusive, reverse: reverse)
+        precondition(axis.rank == 0, "Axis must have rank 0.")
+        precondition(areAxesInRange(axis), "All axis must be in the range `[-rank, rank)`.")
```

Comment:
In `cumprod` and `cumsum`, if the axis is plural then I think the documentation must read, `All axes must be in range ...,` Like the previous conditions.  
If it's a singular value, is calling the function necessary instead of just doing the check locally ?

Reply:
Previously it was given axis instead of axes and also the same comment was written above that `axis` must be in the range `-rank..<rank`. So I followed the same pattern in writing the precondition. Changing the input variable from axis to axes may conflict with some written test cases.

Reply:
Oh then I think, you don't have to right `all axis`, it's just referring to a singular axis. You can start the statement with `Axis must be in range...`

Reply:
I think `rank 0` tensor means that there is only one value in axis tensor. So previously it was written as singular.
`var input: Tensor<Int32> = Tensor(5)`
` input.rank` is giving `0` as output